src/foam/dao/history/HistoryRecord.js

```diff
@@ -0,0 +1,27 @@
+/**
+ * @license
+ * Copyright 2017 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.dao.history',
+  name: 'HistoryRecord',
+  documentation: `Contains an array of property updates`,
+  properties: [
```

Comment:
You need to specify the primaryKey of HistoryRecord. Something like:
ids: [ 'key', 'timestamp' ] (but then you have to make sure there aren't two updates the same ms), or
ids: [ 'key', 'seqNo' ] (and add a seqNo property)